src/monattack.cpp

```diff
@@ -3462,38 +3484,59 @@ void mattack::generator(monster *z, int index)
 void mattack::upgrade(monster *z, int index)
 {
     std::vector<int> targets;
+    int day = calendar::turn / DAYS(1);
     for (size_t i = 0; i < g->num_zombies(); i++) {
         monster &zed = g->zombie(i);
-        if( zed.type->id == "mon_zombie" &&
-            rl_dist( z->pos(), zed.pos() ) <= 5 &&
-            z->sees( zed ) &&
-            z->attitude_to( zed ) != Creature::Attitude::A_HOSTILE ) {
-            targets.push_back(i);
+        // Check this first because it is a cheap check
+        if( zed.type->upgrade_min < day &&
```

Comment:
`upgrade_min` is not scaled to difficulty, neither is `day`. `monster::update_check` however does scale `upgrade_min`.